ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/BlockchainQueries.java

```diff
@@ -87,6 +96,24 @@ public BlockchainQueries(
             ? Optional.of(
                 new TransactionLogBloomCacher(blockchain, cachePath.get(), scheduler.get()))
             : Optional.empty();
+
+    blockBodySoftCache =
+        CacheBuilder.newBuilder()
+            .maximumSize(BLOCK_BODY_SOFT_CACHE_SIZE)
+            .softValues()
```

Comment:
I dno't think there is a connection, softValues is about garbage collection.  If there is memory pressure then the soft references pointing to the value can free the cached values and we have to go back to the DB.  The key always maps to the same instance of a block body, and we don't really care about identity since the values will always be the same.  If we care about the identity of the data objects IMHO we are doing something wrong.

Reply:
It mentions object equality here: https://guava.dev/releases/18.0/api/docs/com/google/common/cache/CacheBuilder.html#softValues()

Reply:
It's not the keys that are identity tested, the values are. I think it only matter for LIFO/LRU eviction.  So I think it's not an issue.